runtime/doc/nvim_clipboard.txt

```diff
@@ -56,4 +56,11 @@ Nvim's X11 clipboard providers only utilize the `PRIMARY` and `CLIPBOARD`
 selections, used for the '*' and '+' registers, respectively.
 
 ==============================================================================
+						   *g:loaded_clipboard_provider*
+
+To disable the clipboard feature, set |g:loaded_clipboard_provider| to 1:
+>
+    let g:loaded_clipboard_provider = 1
+<
```

Comment:
The issue with plugins is rather that we error out already with `has('clipboard')`, but I suppose you're looking into that as well.